### PR TITLE
chore: improve client version handling

### PR DESCRIPTION
**Pull request by igorbernstein2**
* use hardcoded version strings for the version instead of resources. This should improve the version detection when the artifacts get shaded
* log warnings when multiple versions are detected on the classpath
* embed the bigtable-hbase version separately in the user agent to detect mismatches between bigtable-client-core and bigtable-hbase
* port connection tests from master

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableHBaseVersion.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableHBaseVersion.java:2]: 2021

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableHBaseVersion.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hbase.client.AbstractBigtableConnection;
+
+/**
+ * The version in this class is project build version specified in the maven. Maven reads the
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableHBaseVersion.java:30]: nit: "in maven" or "in the maven config"?

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java [hunk 3]
```diff
@@ -454,6 +454,8 @@ private static void setChannelOptions(
     // This information is in addition to bigtable-client-core version, and jdk version.
     StringBuilder agentBuilder = new StringBuilder();
     agentBuilder.append("hbase-").append(VersionInfo.getVersion());
+    agentBuilder.append(", ");
+    agentBuilder.append("bigtable-hbase-").append(BigtableHBaseVersion.getVersion());
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java:458]: My bad, I should have ported https://github.com/googleapis/java-bigtable-hbase/pull/3033/files. Mind if I do that first to reduce the noise on this PR?

**Review by kolea2 — CHANGES_REQUESTED** [review]
This is a bit hard to review - can we separate this out a bit? Can we move all the ports into their own PRs, and remaining changes in this one?

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableHBaseVersion.java (hunk 2 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableHBaseVersion.java:30]: oops, the comment is stale. I just removed it

**Review by kolea2 — APPROVED** [review]

**Pull request merged**